Runtime/Extension/ColliderExtensions.cs

```diff
@@ -19,8 +19,27 @@ public static Transform GetContainingTransform(this Collider collider)
                 return null;
             }
 
-            Rigidbody attachedRigidbody = collider.attachedRigidbody;
+            Rigidbody attachedRigidbody = collider.GetAttachedRigidbody();
             return attachedRigidbody == null ? collider.transform : attachedRigidbody.transform;
         }
+
+        /// <summary>
+        /// Gets the parent <see cref="Rigidbody"/> for the given <see cref="Collider"/> even if the <see cref="GameObject"/> is disabled.
+        /// </summary>
+        /// <param name="collider">The <see cref="Collider"/> to check against.</param>
+        /// <returns>The parent <see cref="Rigidbody"/>.</returns>
+        public static Rigidbody GetAttachedRigidbody(this Collider collider)
+        {
+            Rigidbody attachedRigidbody = collider.attachedRigidbody;
+            if (!collider.gameObject.activeInHierarchy)
+            {
+                foreach (Rigidbody foundRigidbody in collider.GetComponentsInParent<Rigidbody>(true))

```

Comment:
This allocates a new list each time to return to us. Instead we should do what we do elsewhere: Create our own list and reuse it in the call to `GetComponentsInParent` (there's a `bool, List<T>` overload).

Reply:
Is it acceptable here to create a `private static List<Rigidbody>` in the class definition?

Or is making this static going to cause issues as it will become global?

I guess the `GetComponentsInParent` call will always override the list contents each time it is populated though?

Reply:
`static` will be fine and reduce the memory need down to a **single** instance of a list to use. (This is where Unity's restriction to only ever work on the main thread in a single-threaded fashion for all of their callbacks and API allows us to do such things.)